app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt

```diff
@@ -370,6 +376,32 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>(), CancelPledge
         }
     }
 
+    private fun openProjectAndFinish(url: String) {
```

Comment:
Hey just wanna understand this function - is this a test? It looks like it isn't so I'm just wondering how we use the hard-coded `fakeUri` here?

Reply:
> One does not simply open a URL in the browser, especially if that link is identical to a deep link it supports because then it just creates a very sad feedback loop. So, I had to use the same logic in the DeepLinkActivity; by creating an Intent chooser to present the user with options to open the link, not including the app they're already in!

It's not a test. In Android, you can't say "open this URL" if it's a deep link URL. It just tries opening the app again. 

This method uses a `fakeUri` to see which apps can open it. For example, for `http://www.kickstarter.com/` can be opened by Chrome, Kickstarter, and Firefox (I don't have that installed but pretend.) Then I filter out Kickstarter because we're already here. And present the options. 

Let me know if that makes sense. Happy to rename!

Reply:
And for emails: I see now that they're broken for deep linking 🙄

Reply:
Is hard-coding the URI a problem?

Reply:
It's just an arbitrary URI, I could use `http://google.com`. The goal is to see what apps can handle a URI of that structure.

Reply:
ah gotcha